File: backend/src/main/java/com/daewon/backend/entity/Problem.java
package com.daewon.backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Problem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String platform;

    private String problemNumber;

    private String title;

    private String url;

    private String tags;

    public Problem() {
    }

    public Problem(String platform, String problemNumber, String title, String url, String tags) {
        this.platform = platform;
        this.problemNumber = problemNumber;
        this.title = title;
        this.url = url;
        this.tags = tags;
    }

    public Long getId() {
        return id;
    }

    public String getPlatform() {
        return platform;
    }

    public String getProblemNumber() {
        return problemNumber;
    }

    public String getTitle() {
        return title;
    }

    public String getUrl() {
        return url;
    }

    public String getTags() {
        return tags;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public void setProblemNumber(String problemNumber) {
        this.problemNumber = problemNumber;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public void setTags(String tags) {
        this.tags = tags;
    }
}

File: frontend/src/types/problem.ts
/**
 * 백엔드로 문제 저장을 요청할 때 보내는 데이터 구조
 *
 * Spring Boot의 ProblemCreateRequest와 같은 형태
 */
export interface ProblemCreateRequest {
  platform: string;
  problemNumber: string;
  title: string;
  url: string;
  tags: string;
}

/**
 * 백엔드에서 문제를 저장한 뒤 반환하는 데이터 구조
 *
 * 저장 후에는 DB에서 자동 생성된 id가 포함
 */
export interface Problem {
  id: number;
  platform: string;
  problemNumber: string;
  title: string;
  url: string;
  tags: string;
}